### PR TITLE
feat(cpp-shell): make mlog_dump dump plog

### PR DESCRIPTION
**Pull request by acelyc111**
Since Pegasus 2.5.0, the slog has been deprecated entirely, the plog is left.
The 'mlog_dump' tool in cpp-shell is used to dump slog ever, this patch transfer
it to dump plog.

### CODE REVIEW

File: src/shell/commands/debugger.cpp [hunk 1]
```diff
@@ -78,7 +81,7 @@ bool mlog_dump(command_executor *e, shell_context *sc, arguments args)
                                            {0, 0, 0, 0}};
 
     bool detailed = false;
-    std::string input;
+    std::string slog_dir;
```
Comment (empiredan) [inline src/shell/commands/debugger.cpp:84]: Should be `plog_dir`?

File: src/shell/commands/debugger.cpp [hunk 2]
```diff
@@ -101,12 +104,28 @@ bool mlog_dump(command_executor *e, shell_context *sc, arguments args)
...
+        fmt::print(stderr, "ERROR: '{}' is not a directory\n", plog_dir);
+        return false;
+    }
+
+    char splitters[] = {'\\', '/', 0};
+    auto slog_dir_tmp = plog_dir;
+    std::string name =
+        dsn::utils::get_last_component(dirname((char *)slog_dir_tmp.c_str()), splitters);
+    if (name.empty()) {
+        fmt::print(stderr, "ERROR: '{}' is not a valid slog directory\n", plog_dir);
```
Comment (empiredan) [inline src/shell/commands/debugger.cpp:121]:
```suggestion
        fmt::print(stderr, "ERROR: '{}' is not a valid plog directory\n", plog_dir);
```

File: src/shell/commands/debugger.cpp [hunk 3]
```diff
@@ -101,12 +104,28 @@ bool mlog_dump(command_executor *e, shell_context *sc, arguments args)
...
+        return false;
+    }
+
+    char app_type[128];
+    int32_t app_id, pidx;
+    if (3 != sscanf(name.c_str(), "%d.%d.%s", &app_id, &pidx, app_type)) {
+        fmt::print(stderr, "ERROR: '{}' is not a valid slog directory\n", plog_dir);
```
Comment (empiredan) [inline src/shell/commands/debugger.cpp:128]:
```suggestion
        fmt::print(stderr, "ERROR: '{}' is not a valid plog directory\n", plog_dir);
```

File: src/shell/commands/debugger.cpp [hunk 4]
```diff
@@ -101,12 +104,28 @@ bool mlog_dump(command_executor *e, shell_context *sc, arguments args)
...
-        fprintf(stderr, "ERROR: input %s is not a directory\n", input.c_str());
+    if (!dsn::utils::filesystem::directory_exists(plog_dir)) {
+        fmt::print(stderr, "ERROR: '{}' is not a directory\n", plog_dir);
+        return false;
+    }
+
+    char splitters[] = {'\\', '/', 0};
+    auto slog_dir_tmp = plog_dir;
+    std::string name =
+        dsn::utils::get_last_component(dirname((char *)slog_dir_tmp.c_str()), splitters);
```
Comment (empiredan) [inline src/shell/commands/debugger.cpp:119]:
```suggestion
    const auto replica_path = std::filesystem::path(plog_dir).parent_path();
    const auto name = replica_path.filename().string();
```

**Review by GehaFearless — APPROVED** [review]

**Pull request merged**